### ISSUE TITLE
Add option for Dockerfile to inherit command explicitly

### ISSUE THREAD

**Issue by thenewguy** [moby/moby#40811]
When you set the entrypoint in a Dockerfile it clears the CMD set by the parent image.  It would be better if you could reset the CMD with a special "inherit" instruction to allow upstream changes to propagate automatically

**Comment by thaJeztah** [comment]
If an image has both an `ENTRYPOINT` and `CMD`, the `CMD` is used to provide the arguments for `ENTRYPOINT`.

I'm trying to understand your use-case a bit more; could you explain in what situation it would be useful to change the `ENTRYPOINT` to something different, but keep the `CMD` (so, keep the arguments even though the command that they're passed to changed)?

**Comment by thenewguy** [comment]
This particular image is for openresty.  I've added a plugin that requires some setup.  The entrypoint sets some default permissions and such.  However, the command is the same.

Dockerfile:

```
FROM openresty/openresty:centos

EXPOSE 80 443
ENTRYPOINT ["/scripts/docker-entrypoint"]
CMD ["/usr/bin/openresty", "-g", "daemon off;"]

RUN yum install -y gcc make openssl bash curl diff find grep mktemp sed \
    && luarocks install lua-resty-auto-ssl \
    && mkdir /etc/resty-auto-ssl \
    && yum remove -y make gcc \
    && yum install -y openssl bash curl diff find grep mktemp sed \
    && openssl req -new -newkey rsa:2048 -days 3650 -nodes -x509 \
       -subj '/CN=sni-support-required-for-valid-ssl' \
       -keyout /etc/ssl/resty-auto-ssl-fallback.key \
       -out /etc/ssl/resty-auto-ssl-fallback.crt \
    && yum clean all

VOLUME /etc/resty-auto-ssl

COPY files /
```

Entrypoint:
```
#!/usr/bin/env bash

# set global permissions to prevent startup errors - openresty init fixes this
chmod 777 /etc/resty-auto-ssl

exec "$@"
```

**Comment by thaJeztah** [comment]
Ah, gotcha; in your case it's likely because you're bind-mounting these files from the host (so permissions may not match), right?

I can see the use-case; so it would be something like;

```dockerfile
ENTRYPOINT --preserve-cmd ["/scripts/docker-entrypoint"]
```

I'm a bit on the fence to overload the syntax too much for this situation as there's already a way to solve this by copying the `CMD` (which would make it more "transparent" as to what the final result will be). Open for input from @tonistiigi @tiborvass @AkihiroSuda though.

**Comment by thenewguy** [comment]
@thaJeztah exactly, bind mount.  Sometimes commands can be complicated.  This particular one isn't, but it would be pretty handy to be able to just rely on the upstream command working - and if something changes with the command as options are added or whatnot, it would be helpful to avoid unexpected quirks.  Imo something like what you've written isn't particularly complicated to understand

**Comment by thenewguy** [comment]
For example, I've found another one as demonstration:
```
"command": [
	"/usr/local/bin/uwsgi"
	"--strict",
	"--http-keepalive",
	"--add-header=Connection:Keep-Alive",
	"--http-auto-chunked",
	"--http=:80",
	"--https=:443,/etc/pki/tls/certs/server.chained.crt,/etc/pki/tls/private/server.key",
	"--master",
	"--enable-threads",
	"--processes=4",
	"--threads=4",
	"--module=project.wsgi:application",
	"--buffer-size=9216",
	"--thunder-lock"
],
```

It depends on where files are placed in the upstream image.  So those could change and then copy/paste of the command would fail.  But the upstream image would have it correct (or it wouldn't have passed for release)
